evmjit/libevmjit-cpp/Env.cpp

```diff
@@ -62,23 +62,40 @@ extern "C"
 			*o_address = {};
 	}
 
-	EXPORT bool env_call(ExtVMFace* _env, int64_t* io_gas, h256* _receiveAddress, i256* _value, byte* _inBeg, uint64_t _inSize, byte* _outBeg, uint64_t _outSize, h256* _codeAddress)
+	EXPORT bool env_call(ExtVMFace* _env, int64_t* io_gas, int64_t _callGas, h256* _receiveAddress, i256* _value, byte* _inBeg, uint64_t _inSize, byte* _outBeg, uint64_t _outSize, h256* _codeAddress)
 	{
 		auto value = llvm2eth(*_value);
+		auto receiveAddress = right160(*_receiveAddress);
+		auto codeAddress = right160(*_codeAddress);
+		const auto isCall = receiveAddress == codeAddress; // OPT: The same address pointer can be used if not CODECALL
+
+		*io_gas -= _callGas;
+		if (*io_gas < 0)
+			return false;
+
+		if (isCall && !_env->exists(receiveAddress))
+			*io_gas -= static_cast<int64_t>(c_callNewAccountGas); // no underflow, *io_gas non-negative before
+
+		if (value > 0) // value transfer
+		{
+			assert(c_callValueTransferGas > c_callStipend && "Overflow possible");
```

Comment:
It should've been static_assert, but unfortunately these c_* constants are not really compile-time constants because of usage of boost u256 type. By design value transfer gas must be bigger than stipend gas. Despite of that, this assert will fail always or never.